recipes/libsodium/all/conanfile.py

```diff
@@ -54,89 +61,73 @@ def configure(self):
+
+    def build_requirements(self):
+        if self.settings.compiler != "Visual Studio":
+            if self._is_mingw:
+                self.build_requires("libtool/2.4.6")
+            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+                self.build_requires("msys2/cci.latest")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version],
                   destination=self._source_subfolder, strip_root=True)
 
-    @property
-    def _msvc_configuration(self):
-        configuration = ""
-        if self.options.shared:
-            configuration += "Dyn"
-        else:
-            configuration += "Static"
-        build_type = "Debug" if self.settings.build_type == "Debug" else "Release"
-        configuration += build_type
-        return configuration
-
-    @property
-    def _msvc_sln_folder(self):
-        folder = {
-            "14": "vs2015",
-            "15": "vs2017",
-            "16": "vs2019",
-        }.get(str(self.settings.compiler.version), None)
-        if not folder:
-            raise ConanInvalidConfiguration("Unsupported msvc version: {}".format(self.settings.compiler.version))
-        return folder
-
     def _build_msvc(self):
         sln_path = os.path.join(self.build_folder, self._source_subfolder, "builds", "msvc", self._msvc_sln_folder, "libsodium.sln")
-
+        build_type = "{}{}".format(
+            "Dyn" if self.options.shared else "Static",
+            "Debug" if self.settings.build_type == "Debug" else "Release",
+        )
         msbuild = MSBuild(self)
-        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._msvc_configuration)
-
-    @property
-    def _android_abi_str(self):
-        return "androideabi" if str(self.settings.arch) in ["armv6", "armv7"] else "android"
+        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=build_type)
 
     def _configure_autotools(self):
         if self._autotools:
             return self._autotools
         self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        yes_no = lambda v: "yes" if v else "no"
-        if self.settings.os in ("FreeBSD", "Linux"):
-            host = None
-        elif self._is_mingw:
-            host = "{}-w64-mingw32".format("i686" if self.settings.arch == "x86" else self.settings.arch)
-        elif tools.is_apple_os(self.settings.os):
-            host = "{}-apple-{}".format(self.settings.arch, "ios" if self.settings.os == "iOS" else "darwin")
-        elif self.settings.os == "Android":
-            host = "%s-linux-%s" % (tools.to_android_abi(self.settings.arch), self._android_abi_str)
+        if self._is_mingw:
+            self._autotools.libs.append("ssp")
+
+        # TODO: maybe remove this manual host logic? Is it not handled by Autotools helper already?
+        host = None
```

Comment:
yes, it's old code for QNX Neutrino, which were added before conan learnt to recognize these hosts.
should be no longer needed.

Reply:
Correct autotools helper relies on `tools.get_gnu_triplet()` which properly handles all these cases. I've pushed a last commit removing all this logic.

Reply:
okay, less code - less bugs